### ISSUE TITLE
SOAP request fails with UNEXPECTED_INTERNAL_API_ERROR but no response logged for debugging

### ISSUE THREAD

**Issue by cococolov** [googleads/googleads-python-lib#115]
I was trying to use the trialService to create a trial for a campaign, but soap request fails with UNEXPECTED_INTERNAL_API_ERROR, so I asked for help in the API support group.
I was trying to dump debugging information for the engineer who was helping me, but he pointed out the request failed but no response was logged. so he told me to report bug here. 
thanks, and please let me know if you need more information. thanks again.

**Comment by msaniscalchi** [comment]
Hello, thanks for the report.

I'm not able to reproduce the UNEXPECTED_INTERNAL_API_ERROR you're reporting with the add_trial.py example found under campaign_management, so I don't think this is a Client Library issue.

To verify the behavior, I suggest that you modify that sample to accomplish what you're trying to do. I don't expect that you'll continue to see that error, but feel free to let me know if it does and I'll take a closer look.

**Comment by msaniscalchi** [comment]
Following up, it looks like one of the core issues here (as far as this library is concerned) is that you're not seeing the response in your logs. While I can't be absolutely sure due to limited information about your circumstances, this sounds very similar to a [known issue](https://bitbucket.org/jurko/suds/issues/32/logdebug-unicode-encoding-problem-in) with the suds-jurko library.

One feature usually included with suds is the ability to retrieve the last message sent and received. This _could_ have been useful for your purposes, but it was removed from suds-jurko. As another alternative, you can patch in a custom MessagePlugin to add similar functionality.

```
import googleads.adwords
import logging
import suds
from suds.plugin import MessagePlugin
from suds.client import Client


logging.basicConfig(level=logging.INFO)
logging.getLogger('suds.transport').setLevel(logging.DEBUG)


def AddTrial(client):
  # Add trial here...
  pass


def main():
  PatchAdwordsClient()
  client = googleads.adwords.AdWordsClient.LoadFromStorage()

  AddTrial(client)

  # print the last message received.
  print client.plugin.last_received().str()


class MessageStoringPlugin(MessagePlugin):
  def __init__(self):
    self.last_sent_message = None
    self.last_received_message = None

  def sending(self, context):
    if context.envelope:
      self.last_sent_message = context.envelope

  def parsed(self, context):
    if context.reply:
      self.last_received_message = context.reply

  def last_sent(self):
    return self.last_sent_message

  def last_received(self):
    return self.last_received_message


def PatchAdwordsClient():
  googleads.adwords.AdWordsClient.__init__ = PatchedAdWordsClientInit
  googleads.adwords.AdWordsClient.GetService = PatchedAdWordsClientGetService


def PatchedAdWordsClientInit(
    self, developer_token, oauth2_client, user_agent,
    client_customer_id=None, validate_only=False, partial_failure=False,
    https_proxy=None, cache=None):
  self.developer_token = developer_token
  self.oauth2_client = oauth2_client
  self.oauth2_client.Refresh()
  self.user_agent = user_agent
  self.client_customer_id = client_customer_id
  self.validate_only = validate_only
  self.partial_failure = partial_failure
  self.https_proxy = https_proxy
  self.cache = cache
  self.plugin = MessageStoringPlugin()


def PatchedAdWordsClientGetService(self, service_name, version=sorted(googleads.adwords._SERVICE_MAP.keys())[-1],
               server=googleads.adwords._DEFAULT_ENDPOINT):
  if server[-1] == '/': server = server[:-1]
  try:
    proxy_option = None
    if self.https_proxy:
      proxy_option = {
          'https': self.https_proxy
      }

    client = suds.client.Client(
        self._SOAP_SERVICE_FORMAT %
        (server, googleads.adwords._SERVICE_MAP[version][service_name], version, service_name),
        proxy=proxy_option, cache=self.cache, timeout=3600, plugins=[self.plugin])
  except KeyError:
    if version in _SERVICE_MAP:
      raise googleads.errors.GoogleAdsValueError(
          'Unrecognized service for the AdWords API. Service given: %s '
          'Supported services: %s'
          % (service_name, _SERVICE_MAP[version].keys()))
    else:
      raise googleads.errors.GoogleAdsValueError(
          'Unrecognized version of the AdWords API. Version given: %s '
          'Supported versions: %s' % (version, _SERVICE_MAP.keys()))

  return googleads.common.SudsServiceProxy(
      client, googleads.adwords._AdWordsHeaderHandler(self, version))


if __name__ == '__main__':
  main()
```

Of course, this is just a work-around for now. We'll add another patch to the suds-jurko library in a future release that will fix this issue. We'll close this issue when we get around to it.

**Comment by msaniscalchi** [comment]
FYI, this wasn't patched yet. It was prematurely closed. ;-)

**Comment by msaniscalchi** [comment]
This was resolved in 4.6.1, the response is now logged in the event of a WebFault.
